pkg/provisioner/ironic/delete_test.go

```diff
@@ -143,6 +144,27 @@ func TestDelete(t *testing.T) {
 				Value: true,
 			},
 		},
+		{
+			name: "test-power-on",
+			ironic: testserver.NewIronic(t).Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "active",
+					Maintenance:    false,
+					PowerState:     powerOn,
+				},
+			).NodeUpdate(nodes.Node{
+				UUID: nodeUUID,
+			}),
+			powerOn:              true,
+			expectedDirty:        true,
+			expectedRequestAfter: 0,
+			expectedUpdate: &nodes.UpdateOperation{
+				Op:    "replace",
+				Path:  "/maintenance",
+				Value: true,
+			},
+		},
```

Comment:
I think it's missing the test case verifying what happens when the number of retries are greater than `maxPowerOffRetryCount`? (it could be reasonable to have a dedicated separate test for that probably)

Reply:
Please refer to https://github.com/metal3-io/baremetal-operator/pull/816.